testsuite/features/step_definitions/common_steps.rb

```diff
@@ -35,10 +35,18 @@
 
   raise 'SHA256 checksum validation failed' unless validate_checksum_with_file(original_iso_name, iso_path, checksum_path)
 
-  mount_point = "/srv/www/htdocs/#{name}"
-  get_target('server').run("mkdir -p #{mount_point}")
-  get_target('server').run("grep #{iso_path} /etc/fstab || echo '#{iso_path}  #{mount_point}  iso9660  loop,ro,_netdev  0 0' >> /etc/fstab")
-  get_target('server').run("umount #{iso_path}; mount #{iso_path}")
+  if $is_container_provider
+    mount_point = '/srv/www/distributions'
+    get_target('server').run("mkdir -p #{mount_point}")
+    # this needs to be run outside the container
+    get_target('server').run_local("mgradm distro copy #{iso_path} #{name}", verbose: true)
+    get_target('server').run("ln -s #{mount_point}/#{name} /srv/www/htdocs/pub/")
+  else
+    mount_point = "/srv/www/htdocs/#{name}"
+    get_target('server').run("mkdir -p #{mount_point}")
+    get_target('server').run("grep #{iso_path} /etc/fstab || echo '#{iso_path}  #{mount_point}  iso9660  loop,ro,_netdev  0 0' >> /etc/fstab")
+    get_target('server').run("umount #{iso_path}; mount #{iso_path}")
```

Comment:
This way you call server 3 times, I believe it would be better to call one time with full command having broken lines (e.g. using `&&` and `\`)

Reply:
I agree with calling the server 3 times could be improved. I did not refactor this, because it was already this way before my PR and I did not want to do more refactors other than the changes needed.

About duplicating code to make it more readable, I am not that much in favor of it.